### PR TITLE
Master dashboard pga

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
Task: https://www.odoo.com/web#id=10893&view_type=form&model=project.task&menu_id=3942&action=327
Pad: https://pad.odoo.com/p/r.llWYaF3PDnaqtssc

### CODE REVIEW

**Comment by jholze** [comment]
Nice

File: addons/auth_signup/res_users.py [hunk 1]
```diff
@@ -212,7 +213,8 @@ def signup(self, cr, uid, values, token=None, context=None):
             if partner_user:
                 # user exists, modify it according to values
                 values.pop('login', None)
-                values.pop('name', None)
+                if single_email_re.match(partner_user.name) == None:
```
Comment (tde-banana-odoo) [inline addons/auth_signup/res_users.py]: What is the purpose of this change ?

File: addons/auth_signup/views/auth_signup_login.xml [hunk 2]
```diff
@@ -20,7 +20,7 @@
             <div class="form-group field-name">
                 <label for="name" class="control-label">Your Name</label>
                 <input type="text" name="name" t-att-value="name" id="name" class="form-control" placeholder="e.g. John Doe"
-                    required="required" t-att-readonly="'readonly' if only_passwords else None"
+                    required="required" t-att-readonly="'readonly' if (only_passwords and name != login) else None"
```
Comment (tde-banana-odoo) [inline addons/auth_signup/views/auth_signup_login.xml]: What is the purpose of this change ?

File: addons/planner/static/src/js/planner.js [hunk 3]
```diff
@@ -96,6 +96,7 @@
             'click .oe_planner div[id^="planner_page"] a[href^="#planner_page"]': 'change_page',
             'click .oe_planner li a[href^="#planner_page"]': 'change_page',
             'click .oe_planner div[id^="planner_page"] button[data-pageid^="planner_page"]': 'mark_as_done',
+            'hide.bs.modal':'on_close'
```
Comment (tde-banana-odoo) [inline addons/planner/static/src/js/planner.js]: If necessary, those changes should be done in a specific commit / branch. What is the related bug / feature ?

File: addons/planner/static/src/js/planner.js [hunk 4]
```diff
@@ -284,6 +285,9 @@
             }
             self.update_planner(page_id);
         },
+        on_close:function(){
```
Comment (tde-banana-odoo) [inline addons/planner/static/src/js/planner.js]: If necessary, those changes should be done in a specific commit / branch. What is the related bug / feature ?

File: addons/auth_signup/res_users.py (hunk 1 above)
Comment (pga-odoo) [inline addons/auth_signup/res_users.py]: We are inviting users just by `email id` and because of name and log in are required. 
I put same email id in name and provide a way to change the name when user *accept the invitation*
Currently we are not allowing user to change name by popping name from submitted data. So here i allowed name change for user if it's old name was email id.

File: addons/auth_signup/views/auth_signup_login.xml (hunk 2 above)
Comment (pga-odoo) [inline addons/auth_signup/views/auth_signup_login.xml]: here i allowed user to change name if name and log in are same because we inviting user just by email id so we are going provide a way for changing a name when user accept the invitation.

File: addons/planner/static/src/js/planner.js (hunk 3 above)
Comment (pga-odoo) [inline addons/planner/static/src/js/planner.js]: No it is not related to any bug. We haven't use `hide.bs.modal` event because until now we didn't need it
In planner we only having event for `planner_progress_changed` which will triggered when user make change in planner. 
But here i want to make change in Dashboard graph only then when user close the planner, not every time user make change in planner. so i just handled bootstrap's default  `hide.bs.modal` event.

File: addons/web_dashboard/controllers/main.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        return {'installed_apps':installed_apps,'active_users':active_users,'users':users,'planner_data':planner_data,'videos':videos, 'template_user_id': template_user_id}
+
+    @http.route('/dashboard/create_users', type='json', auth='user')
+    def create_user(self, users, optional_message):
+        context = request.context
+        context['create_user'] = 1
+        if optional_message:
+            context['custom_message'] = optional_message
+        user_obj = request.env['res.users'].with_context(context)
+        emails_adderesses= users.strip().strip('\n').split('\n')
```
Comment (jem-odoo) [inline addons/web_dashboard/controllers/main.py]: I would do the email extract client side (in javascript, strip the content of the textarea with spaces and/or comma) and send an array of email to the controllers.

File: addons/web_dashboard/static/src/js/dashboard.js [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+            method: 'xmlid_to_res_id',
+            args: ['base.view_users_form'],
+        }).then(function(res_id) {
+            self.user_form_id = res_id;
+            self.load();
+        });
+    },
+    load:function(){
+        var self = this;
+        ajax.jsonRpc("/dashboard/info", 'call', {}).then(function (data) {
```
Comment (jem-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: Since we are in the WebClient, rather `session.rpc()`

File: addons/web_dashboard/static/src/js/dashboard.js [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+var _t = core._t;
+
+var Dashboard = Widget.extend({
+    template: 'DashboardMain',
+    events: {
+        'click .install_new_apps': 'new_apps',
+    },
+    start:function(){
+        var self = this;
+        ajax.jsonRpc('/web/dataset/call', 'call', {
```
Comment (jem-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: Why don't you use `Model('ir.model.data').call(...)` ?

File: addons/web_dashboard/static/src/js/dashboard.js [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+        this.$el.find('.optional_message_toggler').remove();
+        this.$el.find('textarea.optional_message').slideToggle("fast");
+    }
+
+});
+
+var DashboardPlanner = Planner.PlannerLauncher.extend({
+    template: 'DashboardPlanner',
+     events: {
+        'click .proress_title,.progress': 'setup',
```
Comment (jem-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: `proress_title` ? typo ...

File: addons/web_dashboard/static/src/js/dashboard.js [hunk 9]
```diff
@@ -0,0 +1,256 @@
...
+        this.share_text = encodeURIComponent("I'm using Odoo, an Open-Source Web App that manages my Sales, Projects, Accounting, Website, Warehouse, Shop..and so much more! #imusingodoo");
+        //To-do: add video text and url's here
+        this.video_urls = {
+            'tw':[{'name':"Import your address book",'url':'http://youtu.be/bWXFmIUdasw'},{'name':"Discover the power of filters",'url':'http://youtu.be/epihRRx75Tg'},{'name':"Improve your communication",'url':'http://youtu.be/4__sfWB5nNQ'}],
+            'fb':[{'name':"Build an online quotation",'url':'http://youtu.be/HeDTfbzn_ek'},{'name':"Promote your website",'url':'http://youtu.be/0-Qokk4Ac8A'},{'name':"Boost your contact page",'url':'http://youtu.be/fDQVZ9xJJgI'}],
+            'li':[{'name':"Encode supplier invoices",'url':'http://youtu.be/fA9EEPEJs1E'},{'name':"Plan & follow your activities",'url':'http://youtu.be/naIU-1DrZyk'},{'name':"Give and request legal leaves",'url':'http://youtu.be/4OPVtJ4-dfU'}],
+        };
+        return this._super.apply(this, arguments);
+    },
+    tw:function(){
```
Comment (jem-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: avoid function name that doesn't mean anything. Rather call it `share_twitter`,  `share_facebook`, `share_linkedin`, ....

File: addons/web_dashboard/static/src/js/dashboard.js [hunk 10]
```diff
@@ -0,0 +1,256 @@
...
+    start: function() {
+        var self = this;
+        return self.fetch_application_planner().done(function(apps) {
+            self.planner_apps = apps;
+            return apps;
+        });
+    },
+    setup: function(e){
+        var self = this;
+        this.planner = planner;
```
Comment (jem-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: Where does `planner` come from ?

File: addons/web_dashboard/static/src/js/dashboard.js [hunk 11]
```diff
@@ -0,0 +1,256 @@
...
+        return this._super.apply(this, arguments);
+    },
+    start: function() {
+        var self = this;
+        return self.fetch_application_planner().done(function(apps) {
+            self.planner_apps = apps;
+            return apps;
+        });
+    },
+    setup: function(e){
```
Comment (jem-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: This is not a setup, this function aims to display the planner, right ? So rename it into `display_planner`.

File: addons/web_dashboard/static/src/js/dashboard.js [hunk 12]
```diff
@@ -0,0 +1,256 @@
...
+        this.dialog.$('#PlannerModal').modal('toggle');
+        this.dialog.on("hide.bs.modal", self, function(percent){
+                self.parent.do_reload();
+            });
+    },
+});
+
+var DashboardVideos = Widget.extend({
+    template: 'DashboardVideos',
+     events: {
```
Comment (jem-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: ident

File: addons/web_dashboard/static/src/js/dashboard.js [hunk 13]
```diff
@@ -0,0 +1,256 @@
...
+            self.load();
+        });
+    },
+    load:function(){
+        var self = this;
+        ajax.jsonRpc("/dashboard/info", 'call', {}).then(function (data) {
+            self.$el.find('#apps').append(QWeb.render("DashboardApps", data));
+            new DashboardInvitations(self,data).appendTo('#invitations');
+            new DashboardPlanner(self,data).appendTo('#planner');
+            new DashboardVideos(self,data).appendTo('#videos');
```
Comment (jem-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: You are passing `data` to your 3 sub-widgets, but they don't use `data`. Even the planner widget re-fetch planner data with `fetch_application_planner`. This is performance killer. Privilege one ONE rpc call to `/dashboard/info` to get ALL the information your need for your widget, and give only the required information to your widget constructor (`new DashboardInvitations(self,data['users']) .... ew DashboardVideos(self,data['planners']) ....`)

File: addons/web_dashboard/static/src/less/dashboard.less [hunk 14]
```diff
@@ -0,0 +1,136 @@
+@bg-main:#F2F2F2;
+
+.dashboard {
```
Comment (jem-odoo) [inline addons/web_dashboard/static/src/less/dashboard.less]: prefix all your CSS class with `o_web_dashboard_`
Use maximum native bootstrap class. You have too much less.

File: addons/web_dashboard/static/src/js/dashboard.js [hunk 15]
```diff
@@ -0,0 +1,256 @@
...
+            });
+        }else{
+            this.do_warn(_t("Please enter email addresses"),"");
+        }
+
+    },
+    on_user_clicked: function (event) {
+        // event.preventDefault();
+        var user_id = $(event.target).data('user-id');
+
```
Comment (jem-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: remove your commented line, an useless spaces

File: addons/web_dashboard/static/src/js/dashboard.js [hunk 16]
```diff
@@ -0,0 +1,256 @@
...
+    },
+    new_apps:function(){
+        this.do_action('base.open_module_tree', {
+            'additional_context': {'search_default_app':1,'search_default_not_installed':1}
+        });
+    },
+    do_reload:function(){
+        this.$el.find('.col > div').children().remove();
+        this.load();
+        $('.loader').addClass("hidden");
```
Comment (jem-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: There is already a loader in Odoo. Why adding a second one ? What is its specialty ?

File: addons/planner/static/src/js/planner.js [hunk 17]
```diff
@@ -288,6 +289,9 @@ var PlannerDialog = Widget.extend({
         }
         self.update_planner(page_id);
     },
+    on_close:function(){
+            this.trigger("hide.bs.modal");
```
Comment (jem-odoo) [inline addons/planner/static/src/js/planner.js]: rebase your branch and remove this change. Planner has change, there is another way to do that ;)

File: addons/web_dashboard/controllers/__init__.py [hunk 18]
```diff
@@ -0,0 +1 @@
+import main
```
Comment (jem-odoo) [inline addons/web_dashboard/controllers/__init__.py]: add the coding header

File: addons/web_dashboard/static/src/js/dashboard.js [hunk 19]
```diff
@@ -0,0 +1,256 @@
+odoo.define('dashboard.dashboard', function (require) {
```
Comment (jem-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: define `web.dashboard` to stay coherent with your module name

File: addons/web_dashboard/controllers/main.py [hunk 20]
```diff
@@ -0,0 +1,68 @@
+from openerp import tools
+from openerp.addons.web import http
+from openerp.addons.web.http import request
```
Comment (jem-odoo) [inline addons/web_dashboard/controllers/main.py]: Import the native object, not the "already imported one"
``
from openerp import http, tools
from openerp.http import request
``

File: addons/web_dashboard/static/src/js/dashboard.js [hunk 21]
```diff
@@ -0,0 +1,256 @@
...
+    on_user_clicked: function (event) {
+        // event.preventDefault();
+        var user_id = $(event.target).data('user-id');
+
+        var action = {
+            type:'ir.actions.act_window',
+            view_type: 'form',
+            view_mode: 'form',
+            res_model: 'res.users',
+            views: [[this.user_form_id, 'form']],
```
Comment (jem-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: Why do you need the `user_form_id` ?

**Comment by jem-odoo** [comment]
Please rebase your branch. Planner has changed (from `planner` to `web_planner` module).
Also have a look to the guidelines (https://www.odoo.com/documentation/8.0/reference/guidelines.html) to correctly name your variables, files, ...
